File: Shared/Dto/ForWebApi/Trace/TraceStateDto.cs
﻿using System;
using System.Collections.Generic;

namespace Iit.Fibertest.Dto
{
    public class TraceStateDto
    {
        public TraceHeaderDto Header = new TraceHeaderDto();
        public FiberState TraceState;
        public BaseRefType BaseRefType;
        public EventStatus EventStatus;
        public string Comment;
        public DateTime MeasurementTimestamp;
        public DateTime RegistrationTimestamp;
        public int SorFileId;
        public List<AccidentLineDto> Accidents = new List<AccidentLineDto>();
        public bool IsLastStateForThisTrace;
        public bool IsLastAccidentForThisTrace;
    }
}

File: Client/WebClient/src/app/models/dtos/trace/traceStateDto.ts
import { TraceHeaderDto } from "./traceHeaderDto";
import { BaseRefType } from "../../enums/baseRefType";
import { EventStatus } from "../../enums/eventStatus";
import { FiberState } from "../../enums/fiberState";
import { AccidentLineDto } from "./accidentLineDto";

export class TraceStateDto {
  header: TraceHeaderDto = new TraceHeaderDto();
  rtuId: string;
  traceId: string;
  traceState: FiberState;
  baseRefType: BaseRefType;
  eventStatus: EventStatus;
  comment: string;
  measurementTimestamp: Date;
  registrationTimestamp: Date;
  sorFileId: number;
  accidents: AccidentLineDto[];
  isLastStateForThisTrace: boolean;
  isLastAccidentForThisTrace: boolean;
}
